### ISSUE TITLE
Noise parameter `Sigma` can't be recovered from `PosteriorGPs`

### ISSUE THREAD

**Issue by Crown421** [JuliaGaussianProcesses/AbstractGPs.jl#368]
I noticed that once you have a `PosteriorGP`, it is no longer possible to recover the noise level that was given in the construction of the `FiniteGP`. 

My quick fix for this would be to just add `Sigma_y` to the `data` field in the `PosteriorGP`. I have this currently implemented in a local copy, and am happy to make a quick PR.

**Comment by willtebbutt** [comment]
Hmm. I see where you're coming from, but I'm reluctant to add redundant fields into the `PosteriorGP` object. Is there a reason that it's hard to just keep the noise variance around in whatever your particular context is / a reason that it _must_ go into the posterior object?

**Comment by Crown421** [comment]
Not quite. The decoupled sampling/ matheron's rule works of generating samples from the noise distribution, so `Normal(0, Sigma_y)`. 
At least that is the case with the standard implementation
![image](https://github.com/JuliaGaussianProcesses/AbstractGPs.jl/assets/9107200/9d221268-6c40-4776-bb58-fb4c95b12973)
(from https://arxiv.org/pdf/2002.09309.pdf)

I can't think of a way to do this without knowing the observation noise.

**Comment by theogf** [comment]
But isn't (Knn + sigma)^-1 obtainable by just taking the inverse of the prior covariance?

**Comment by willtebbutt** [comment]
Ahhh, I see what you mean. I agree that you need access to the noise model when doing the pathwise sampling stuff. I think I also agree that the right time to introduce the approximations involved is when you've got hold of a `PosteriorGP` object (since the pathwise sampling stuff is really introducing an approximate posterior).

I'd definitely be happy to review this PR.
